File: linked_list/introduction.c
#include <stdio.h>
#include <stdlib.h>

struct Node{
	int data;
	struct Node *next;
};

void printList(struct Node *n){
	while (n != NULL){
		printf("%d\n",n->data );
		n = n->next;
	}
}
// Program to create simple linked list with 3 nodes 
int main()
{
	struct Node* head = NULL;
	struct Node* second = NULL;
	struct Node* third = NULL;

	// allocates 3 node int the heap
	head = (struct Node*) malloc(sizeof(struct Node));
	second = (struct Node*) malloc(sizeof(struct Node));
	third = (struct Node*) malloc(sizeof(struct Node));

	head->data = 1;
	head->next = second;
	second->data = 2;
	second->next = third;
	third->data = 3;
	third->next = NULL;
	printList(head);
	return 0;

}

File: linked_list/introduction.py
# A simple program to introduce a linked list

# Node class
class Node:

	# Function to initialize node object
	def __init__(self, data):
		self.data = data # assigns data
		self.next = None # Inotialize next as null

#Linked List class
class LinkedList:
	def __init__(self):
		self.head = None

	def printList(self):
		temp = self.head
		while(temp):
			print temp.data
			temp = temp.next

# Code execution start here.
if __name__ == '__main__':

	llist = LinkedList()

	llist.head = Node(1)
	second = Node(2)
	third = Node(3)

	llist.head.next = second
	second.next = third
	llist.printList()
	
